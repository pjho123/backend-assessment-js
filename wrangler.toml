name = "ma-project"
main = "src/index.ts"
compatibility_date = "2023-04-18"
workers_dev = false
compatibility_flags = ["experimental:nodejs_compat_v2"]

[vars]
PGHOST = "ep-cold-meadow-a78on0ae-pooler.ap-southeast-2.aws.neon.tech"
PGDATABASE = "assessment-db"
PGUSER = "assessment-db_owner"
PGPASSWORD = "CnkBFR9sa2df"
ENDPOINT_ID = "ep-cold-meadow-a78on0ae-pooler"
API_KEY = "c6c687a0a7209f877f89a40824907ae5"