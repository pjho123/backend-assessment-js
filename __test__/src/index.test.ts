import { unstable_dev } from "wrangler";
import type { UnstableDevWorker } from "wrangler";
import { describe, expect, it, beforeAll, afterAll } from "vitest";

describe("Worker", () => {
	let worker: UnstableDevWorker;

	beforeAll(async () => {
		worker = await unstable_dev("src/index.ts", {
			experimental: { disableExperimentalWarning: true },
		});
	});

	afterAll(async () => {
		await worker.stop();
	});

	it("should return 404 not found if route is invalid", async () => {
		const resp = await worker.fetch("http://localhost:8787/api/invalid");

		const jsonResponse = await resp.json();
		expect(jsonResponse).toMatchSnapshot();
	});
});
