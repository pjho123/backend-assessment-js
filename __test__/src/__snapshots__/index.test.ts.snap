// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Worker > should return 404 not found if route is invalid 1`] = `
{
  "error": "Not Found",
  "status": 404,
}
`;
